crates/stages/stages/src/stages/s3/downloader/worker.rs

```diff
@@ -0,0 +1,69 @@
+use reqwest::{blocking::Client, header::RANGE};
```

Comment:
why do we use blockin client here?
blocking is less performant than the async one, we're not constraint by async so I feel like this entire thing should be async